### ISSUE TITLE
Render additional component below Index list

### ISSUE THREAD

**Issue by nickjvm** [grommet/grommet-index#54]
We have a need to display supplementary content below an index list in some cases. (Similar to `emptyMessage`, but we also need it to show when there are a few items in the list as well.)

I'm thinking of implementing a `footer` prop that takes any node and is rendered anytime the prop is defined, but wanted to get suggestions before submitting a PR. 

``` jsx
<Index footer={<Box pad="medium">Supplementary content</Box>} ... />
```

![image](https://cloud.githubusercontent.com/assets/4998403/18210931/3a6267d2-70f8-11e6-9a23-22ecc1ac68e4.png)

One thing I'm not sure about this approach is how it will affect the `onMore` infinite scrolling feature. It would seem that you wouldn't want to use both at the same time, because potentially you'll never see the footer if additional items being loaded are always pushing it down. Also, infiniate scroll relies on the user scrolling to the bottom of the document. Depending on the height of the footer, the user may see the loading icon and wait for something to happen without having scrolled far enough to trigger the `onMore` function.

Here's a codepen of what I'm considering.
http://codepen.io/nickjvm/pen/vXYAKx?editors=0010

**Comment by alansouzati** [comment]
Good point. I agree with the fact that you would not want to use footer and onMore together.

Maybe we can add a console.warn inside `componentDidMount`  where it will warn developers about this dangerous combination ?

**Comment by nickjvm** [comment]
This is resolved by #55
